include/shell/shell.h

```diff
@@ -485,6 +485,7 @@ enum shell_state {
 	SHELL_STATE_UNINITIALIZED,
 	SHELL_STATE_INITIALIZED,
 	SHELL_STATE_ACTIVE,
+	SHELL_STATE_HALTED,
```

Comment:
What is purpose of this new state?

Reply:
Agreed, I think at this point a state diagram/transition list would be helpful. Can we only move to halted from active? Where can we go from the various panic modes? etc.

Reply:
Removed.
> Where can we go from the various panic modes? 

Panic modes are entered when `LOG_PANIC` is called and shell as a logger backend is informed about panic. `LOG_PANIC` is called on unrecoverable system fault. There is no way back from panic.